indra/newview/llviewershadermgr.cpp

```diff
@@ -430,13 +430,23 @@ void LLViewerShaderMgr::finalizeShaderList()
     mShaderList.push_back(&gDeferredDiffuseProgram);
     mShaderList.push_back(&gDeferredBumpProgram);
     mShaderList.push_back(&gDeferredPBROpaqueProgram);
-    mShaderList.push_back(&gGLTFPBRMetallicRoughnessProgram);
+
+    if (gSavedSettings.getBOOL("GLTFEnabled"))
+    {
+        mShaderList.push_back(&gGLTFPBRMetallicRoughnessProgram);
+    }
+
     mShaderList.push_back(&gDeferredAvatarProgram);
     mShaderList.push_back(&gDeferredTerrainProgram);
-    for (U32 paint_type = 0; paint_type < TERRAIN_PAINT_TYPE_COUNT; ++paint_type)
+
+    if (gSavedSettings.getBOOL("LocalTerrainPaintEnabled"))
     {
-        mShaderList.push_back(&gDeferredPBRTerrainProgram[paint_type]);
+        for (U32 paint_type = 0; paint_type < TERRAIN_PAINT_TYPE_COUNT; ++paint_type)
+        {
+            mShaderList.push_back(&gDeferredPBRTerrainProgram[paint_type]);
```

Comment:
This is not the terrain bake program. This should be `gPBRTerrainBakeProgram`

Reply:
Whoopsie... good catch!